### ISSUE TITLE
opacity of live drawing

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#981]
we have lost the possibility to adjust the opacity of live drawing (we used to have this in the old UpStage & i think also in the new platform, before we separated out live and object drawing). can this be implemented?

i envisage a slider similar to the one for the size of the brush, that allows me to set the opacity of the line before i start drawing.
@vickis @elisadcg do you have any comments on how this could best be incorporated into the toolbar?

**Comment by helenvarley** [comment]
this would also be good to have for the object drawing, wouldn't it? so that one object drawing could have lines of different opacity within it

**Comment by vickis** [comment]
yes - would be great to have a slider in the toolbar for opacity (for live
drawing)
the work around is to draw the object and then change the opacity on the
stage (not as fluid for sure...

**Comment by tunguyentk** [comment]
![Screenshot 2022-12-07 153658](https://user-images.githubusercontent.com/117065175/206129220-acc9e97d-aff4-418a-8cde-033997c1e146.png)
@vickis @helenvarley  i am recreating a stroke with a given opacity , this is really hard so i did some trick , but it removes the possibility of mixing colors between strokes with opacity ( it means strokes with opacity will have a certain penetration and it will show the stroke below it, this will create dithering if the 2 strokes overlap, but my way removed it). Does this removal affect the drawing?

**Comment by vickis** [comment]
is this on dev1 already - I tried to test it but could not work out how to
change opacity
I think if is not allowing the other colour to be shown through (or
anything even?) then it is not really achieving opacity

its a 'would be great to have' but also if tricks are involved how does
that stack up for others coming into the code (i sm probably just
demonstrating my ignorance here :)

thanks for trying!

**Comment by vickis** [comment]
apologies - i mean it is not achieving transparency (or a scale of opacity)

I suggest if any work around does not comply with straight forward
understandable coding it should be deemed not workable

This can remain on the 'great to have' list for future attention

Thanks vicki :)

**Comment by tunguyentk** [comment]
@vickis My way is generating quite a few problems so I haven't deloyed it into devapp yet. It's quite difficult to create more functionality but keep everything from before working :))

**Comment by vickis** [comment]
ok - well in that case let's make opacity something to research in future :)
adding anything that compromises the main code is problematic

thanks for your work so far though!
vicki :)

**Comment by gloriajw** [comment]
I think he means it compromises previously opaque images. It likely isn't
related to code. Do you mind reproducing anything opaque after he applies a
fix?

**Comment by vickis** [comment]
Oh sure ok
I would be happy to play with whatever the solution is to affect drawing
opacity on dev app
Sorry for my misunderstanding ;)

**Comment by tunguyentk** [comment]
@gloriajw  I think I will start from begin Looks like my way is pretty bad and it doesn't work really well. Creating opacity is completely different from resizing or changing color. It seems quite difficult to me.

**Comment by gloriajw** [comment]
@tunguyentk Thank you. Take your time, and try adding/removing one feature at a time, then test. Approach it slowly and methodically. Let me know how it goes, or if you need help.

**Comment by helenvarley** [comment]
just to clarify: this issue applies to live drawing only, as we can adjust the opacity of object drawings after they have been saved. 
previously in UpStage there was an opacity slider, similar to the slider for the size of the brush, in the drawing palette. can this be added as another thing to set in the live drawing toolbar, along with the colour and the size of the brush? if we can select a colour there, surely it's not any more difficult to set the opacity of the colour?

**Comment by vickis** [comment]
Previously we could not save and change opacity as a function of the object - so had a slider to affect the drawn image - is the issue that the colour picker does not allow opacity *other tools are more functional to the canvas eg brush size, undo clear etc ...

**Comment by helenvarley** [comment]
no, the issue is that with live drawing there is no possibility to adjust the opacity. this would be another slider, similar to the one that allows us to set the size of the brush, that would set the opacity of the live drawing line.

**Comment by vickis** [comment]
Yes :) 
Wondering if the opacity is tied to colour sort 
Be great if available as a separate parameter :) 
9 Jun 2024 at 10:15 by ***@***.***:

**Comment by helenvarley** [comment]
i've added @webdev604 to this issue. it's not a biggie & it's not urgent, but if it's not difficult it would be nice to have.

we just need another slider in the live drawing toolbar, i'm envisaging it above or below the brush size slider, to adjust the opacity of the line before it's drawn. it should look the same as the opacity slider for backdrops.

here's a mock-up of what the toolbar could look like (might have to be slightly longer to accommodate the two sliders):

![opacity_slider_live_drawing](https://github.com/upstage-org/upstage/assets/3437324/5e297830-f4bc-4037-bbf7-ef2dc5973c45)

**Comment by helenvarley** [comment]
is it possible to implement this in this release? if it's too much then we can postpone it.

**Comment by gloriajw** [comment]
Should be back, and should work correctly.

**Comment by helenvarley** [comment]
great! would be good for @vickis and other expert drawers to test this, i'll ask them on thursday
